### PR TITLE
Understanding polymorphism & abstract classes using C#

### LINKED ISSUE #2520: Understanding polymorphism & abstract classes using C#
### Proposed title of the tutorial.
Understanding polymorphism & abstract classes using C#.

### Introduction paragraph(s):

**Polymorphism** is often referred to as the third pillar of object-oriented programming, after encapsulation and inheritance. *Polymorphism* is a Greek word that means "many-shaped". At run time, objects of a derived class may be treated as objects of a base class in places such as method parameters and collections or arrays. When this polymorphism occurs, the object's declared type is no longer identical to its run-time type.

**Polymorphism** is a very helpful concept when it comes to a single abstract idea used in different ways and shapes.
With polymorphism, you can define a single abstract class and reshape it in different ways to suit your methods & ideas. 

### Key takeaways:

- Understand the concept of polymorphism
- Understand abstract classes
- Multiple examples of polymorphism

### References:

https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/object-oriented/polymorphism

### PR DESCRIPTION
**Pull request by mohamedgh16**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes #2520

### CODE REVIEW

**Comment by hectorkambow** [comment]
@mohamedgh16 
👋 Good afternoon - Hector here.
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

Not sure if its boilerplate code. 🤔 


[Understanding polymorphism & abstract classes using C# #2699.pdf](https://github.com/section-engineering-education/engineering-education/files/6771486/Understanding.polymorphism.abstract.classes.using.C.2699.pdf)

File: content/articles/understanding-polymorphism-abstract-classes-using-c-/index.md [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+- A basic understanding of the C# programming language.
+- A basic understanding of Inheritance in C# programming language.
+
+If you're new to Inheritance, you should check this tutorial that would help you understand the concept of it [here](https://www.section.io/engineering-education/getting-started-with-inheritance-using-c/).
+
+### Abstract classes
+
+Abstract classes are too generic to be instantiated and are used as base classes. The definitions for abstract classes are not complete. The derived classes must be defined as the missing pieces and should contain abstract methods or abstract properties.
+
+The following example will explain how we can define an abstract class to make a Payroll System Using Polymorphism.
```
Comment (zolomohan) [inline content/articles/understanding-polymorphism-abstract-classes-using-c-/index.md:24]: It seems like you can provide a detailed explanation of what abstract classes & abstract methods are before jumping into the example.

**Review by zolomohan — CHANGES_REQUESTED** [review]
@mohamedgh16 Please go through the below suggestion and we can move to the final review.

**Comment by zolomohan** [comment]
Also, see if you can reduce the plagiarism percentage.

**Comment by mohamedgh16** [comment]
@hectorkambow @zolomohan Done! 👏👀

**Comment by ahmadmardeni1** [comment]
Is this one ready for a final review? @zolomohan

**Comment by zolomohan** [comment]
@ahmadmardeni1 Just have to add the front matter. Will update soon.

**Comment by zolomohan** [comment]
@ahmadmardeni1 it's good to go.

**Comment by ahmadmardeni1** [comment]
Good work 🚀@mohamedgh16 @zolomohan

**Pull request merged**